### PR TITLE
TOOL-10581 linux-pkg should be able to use git token to push to remote

### PR DESCRIPTION
**Pull request by pzakha**
This now allows the auto-update logic to push changes to the remote using a git token. Previously only a combination of a user & password were supported.

GitHub is deprecating the usage of user:password auth for some of its APIs (https://developer.github.com/changes/2020-02-14-deprecating-password-auth/). It's unclear if `git push` is affected by this, but in any case using a token is a much better security practice.

## Next Steps
Update Jenkins to pass a token instead of a user & password (see TOOL-10580).

## Testing
Tested by running `./sync-with-upstream.sh bpftrace` manually on a bootstrap VM, targetting my developer fork of the `bpftrace` repo.
- Tested passing PUSH_GIT_TOKEN in env
- Tested passing PUSH_GIT_USER & PUSH_GIT_PASSWORD in env
- Tested not passing anything in env and getting prompted for a token

### CODE REVIEW

**Comment by pzakha** [comment]
cc: @grodr

**Review by sdimitro — APPROVED** [review]

File: lib/common.sh [hunk 1]
```diff
@@ -857,19 +857,17 @@ function merge_with_upstream_default() {
...
-	if [[ -z "$PUSH_GIT_USER" ]] || [[ -z "$PUSH_GIT_PASSWORD" ]]; then
+	if [[ -z "$PUSH_GIT_TOKEN" ]] && [[ -z "$PUSH_GIT_USER" || -z "$PUSH_GIT_PASSWORD" ]]; then
 		if [[ -t 1 ]]; then
 			if [[ "$DRYRUN" == "false" ]]; then
 				echo_bold "WARNING: this is NOT a dry-run, you are pushing to" \
 					"a production branch"
 			fi
-			echo "Please enter git credentials to push to remote ($DEFAULT_PACKAGE_GIT_URL)."
-			read -r -p "Username: " PUSH_GIT_USER
-			read -r -s -p "Password: " PUSH_GIT_PASSWORD
```
Comment (grodr) [inline lib/common.sh:868]: Will this section still be needed in the case that PUSH_GIT_USER and PUSH_GIT_PASSWORD are passed in?

**Review by grodr — APPROVED** [review]
Other than than the one comment I left, LGTM! Ship it!

File: lib/common.sh (hunk 1 above)
Comment (pzakha) [reply lib/common.sh:868]: This whole section (`if [[ -t 1 ]]; then`) is only for when we run the scripts manually, in which case none of the PUSH_* variables will usually be passed.

Comment (grodr) [reply lib/common.sh:868]: Ah, okay, makes sense.

**Review by sebroy — APPROVED** [review]

**Pull request merged**